src/Builder/BuilderEncoder.php

```diff
@@ -206,6 +211,10 @@ private function encodeAsDollarObject(OperatorInterface $value): stdClass
      */
     private function encodeAsSingle(OperatorInterface $value): stdClass
     {
+        if ($value instanceof BackedEnum) {
+            return $this->wrap($value, $value->value);
```

Comment:
Is this code path used? Your earlier [comment](https://github.com/mongodb/mongo-php-builder/pull/38#discussion_r1457938643) suggested that only `encodeAsArray()` would need to check for a BackedEnum instance.

Reply:
Indeed, this code path is not used for now.